tests/python_package_test/test_dask.py

```diff
@@ -1255,3 +1255,59 @@ def test_sklearn_integration(estimator, check, client):
 def test_parameters_default_constructible(estimator):
     name, Estimator = estimator.__class__.__name__, estimator.__class__
     sklearn_checks.check_parameters_default_constructible(name, Estimator)
+
+
+@pytest.mark.parametrize('task', ['binary_classification', 'multi-class_classification', 'regression'])
+@pytest.mark.parametrize('output', data_output)
+def test_predict_with_raw_score(task, output, client):
+    if task.endswith('classification'):
+        objective = 'classification'
+        model_factory = lgb.DaskLGBMClassifier
+    elif task == 'regression':
+        objective = 'regression'
+        model_factory = lgb.DaskLGBMRegressor
+
+    if task.startswith('multi'):
+        n_centers = 3
+    else:
+        n_centers = 2
+    _, _, _, dX, dy, _ = _create_data(
+        objective=objective,
+        output=output,
+        centers=n_centers
+    )
+    params = {
+        'client': client,
+        'n_estimators': 1,
+        'num_leaves': 2,
+        'time_out': 5,
+        'min_sum_hessian': 0
+    }
+    model = model_factory(**params)
+    model.fit(dX, dy)
+    raw_predictions = model.predict(dX, raw_score=True).compute()
+    computed_mean = np.mean(raw_predictions, axis=0)
+
+    def compute_expected_predictions_mean(tree_df: pd.DataFrame):
+        """Computes the expected mean of the predictions from the dataframe of a tree with two leaves."""
+        leafs_df = tree_df.iloc[1:]  # remove root node
+        leafs_weights = leafs_df['weight'].values
+        leafs_values = leafs_df['value'].values
+        return leafs_values @ leafs_weights / leafs_weights.sum()
```

Comment:
```suggestion
    def compute_expected_predictions_mean(tree_df: pd.DataFrame):
        """Computes the expected mean of the predictions from a model with one tree and two leaves."""
        leaf_values = tree_df['value']
        leaf_weights = tree_df['weight']
        sum_weights = leaf_weights[1] + leaf_weights[2]
        return leaf_values[1] * (leaf_weights[1]/sum_weights) + leaf_values[2] * (leaf_weights[2]/sum_weights)
```

Unless you disagree with my suggestion from https://github.com/microsoft/LightGBM/issues/3793#issuecomment-785613951 because you see a correctness or performance issue, please just use that code or something similarly straightforward.

I'm opposed to adding uncommon operators like [the `@` operator](https://www.python.org/dev/peps/pep-0465/)  unless there is a correctness of performance reason to do so. In my opinion, doing that makes this code harder to understand for other contributors.